boards/arm/disco_l475_iot1/disco_l475_iot1.dts

```diff
@@ -81,6 +81,7 @@
 	hts221@5f {
 		compatible = "st,hts221";
 		reg = <0x5f>;
+		drdy-gpios = <&gpiod 15 GPIO_ACTIVE_HIGH>;
```

Comment:
@avisconti drdy-gpio is described as not required in binding (cf `dts/bindings/sensor/st,hts221.yaml`)
```
properties:
    drdy-gpios:
      type: phandle-array
      required: false
      description: DRDY pin
```
If not required, it should not produce error if not present.
So, maybe this change in the board is correct, but one more fix is required:
-either this property should be made `required: true`
-either hts221 driver should not produce build error when not present

Reply:
@erwango 
I think it would be better thesecond option (i.e. change the driver). But (and I had a similar question in a recent PR) I don't know if it is possible to check in the driver if any gpio is defined or not. Currently I just check CONFIG_HTS221_TRIGGER.

I explain better. We have macros like "DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)" that can be used in the driver to see if in the DT we have defined instances on SPI bus. And this is helpful for enable/disable part of code in the driver.
In case of GPIO what would be the best way to check? Maybe if DT_INST_GPIO_LABEL(inst, drdy_gpios) is NULL? But if there are multiple instances we would require to have a macro with _ANY_INST_. Do you see my point?

If we want to fix it quickly we can also remove HTS221 from the build_all trigger file. But I guess that sooner or later we need to have some special macro as described above (if not yet present)

Reply:
@avisconti you can use the following DT macro to hcekc presence of  an optional dt prop: `DT_INST_NODE_HAS_PROP`.
You can check examples of usage in the tree.

Reply:
@erwango 
OK, I'll do in this way as it seems more correct. I will leave the DTS modification for disco_l475_iot1 for a later change, as it is a totally different context (by the way I do not think it is strictly required drdy for a humidity sensor, so we may want to carefullt evaluate it).

By the way I guess that if we require to make hts221 multi-instance we need a more complex macro with ANY in it (Does ANY hts221 instance has this PROP?). Do you agree?